userspace/libsinsp/user_event.cpp

```diff
@@ -20,6 +20,57 @@ along with sysdig.  If not, see <http://www.gnu.org/licenses/>.
 #include "sinsp_int.h"
 #include "user_event.h"
 
+const event_scope::string_list_t event_scope::RESERVED_STRINGS =
+	{"'"/*!must be first!*/, "=", " and "};
+const event_scope::string_list_t event_scope::REPLACEMENT_STRINGS =
+	{"\\'", "='", "' and "}; // !do not change order!
+
+string&& event_scope::escape(std::string&& scope)
+{
+	trim(scope);
+	string_list_t::const_iterator res_it = RESERVED_STRINGS.cbegin();
+	string_list_t::const_iterator res_end = RESERVED_STRINGS.cend();
+	string_list_t::const_iterator rep_it = REPLACEMENT_STRINGS.cbegin();
+	string_list_t::const_iterator rep_end = REPLACEMENT_STRINGS.cend();
+	for(; res_it != res_end && rep_it != rep_end; ++res_it, ++rep_it)
+	{
+		replace_in_place(scope, *res_it, *rep_it);
+	}
+	scope.append(1, 0x27); // terminating single quote
+	return std::move(scope);
+}
+
+// utility function to check that scope entry is valid;
+// valid entries can not contain '=' character or " and " string
+bool event_scope::check(const std::string& scope)
+{
+	string_list_t::const_iterator rs = RESERVED_STRINGS.cbegin();
+	++rs;
+	for(; rs != RESERVED_STRINGS.end(); ++rs)
+	{
+		if(scope.find(*rs) != std::string::npos)
+		{
+			return false;
+		}
+	}
+	return true;
+}
+
+bool event_scope::assemble(std::string& scope, const std::string& name, const std::string& value)
+{
+	if(check(name) && check(value))
+	{
+		if(scope.length()) { scope.append(" and "); }
+		scope.append(name).append(1, '=').append(value);
```

Comment:
In my opinion, a truly proper solution to this is to have separate `event_scope` class that takes care of everything related to even scopes. I can spend some time on extracting this into a separate class.